### DISCUSSION TITLE
How to do calculate with dependency

### DISCUSSION THREAD

**Discussion by hengyoush** [graphql-java/graphql-java#3221] [Q&A]
i have the following type definition:
```
type Range {
	count :Int
	fails: Int
	failPercent: Float
}
```
I want to calculate `failPercent = fails / count`, how can i do it? can i control the  DataFetcher execution order ？

**Comment by bbakerman** [comment]
graphql does allow calculations to be expressed.

Unlike say SQL (which allows you to say what data you want back AND that calculate things `select (count / fails) as fPerc`

You can only select fields.

So your `failPercent` needs to have a data fetcher behind it that does that caculation

However each field gets a "source" object passed into it

you source object probabably looks logicall like

```
{
   count : 300
   fails : 10
   }
```

and so your data fetcher might be (in rough psdeudo code)

```
DataFecther percentFailDF = env -> {
   MySourceOfData data = env.getSource()
   return data.getFails() / data.getCount();
}

```

I dont know what "source" object you pass in but it needs to  something like this.  Then the order of fields does not matter.

**Reply by hengyoush** [reply]
Got it thanks
